### PR TITLE
For issue #13 When playing against a bot, human is always the first player.

### PR DESCRIPTION
**Pull request by percevalve**
Bot move was made after a POST request was made.

Created function _make_bot_move in views.py and added a condition on the first player to test if not human when the board is created.

Also have some changes to allow setting specific Postgres setting, let me know if it should be in a different PR.

### CODE REVIEW

File: server/.env-sample [hunk 1]
```diff
@@ -5,6 +5,11 @@ BOTANY_OPCODE_LIMIT = 10_000
 
 DEBUG = yes
 
+
+POSTGRES_USER = user
+POSTGRES_PASSWORD = password
+POSTGRES_URL = db_url
```
Comment (mjpieters) [inline server/.env-sample]: This should not be in this changeset, really.

File: server/.gitignore [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+media
+
+# If your build process includes running collectstatic, then you probably don't need or want to include staticfiles/
+# in your Git repository. Update and uncomment the following line accordingly.
+# <django-project-name>/staticfiles/
+
+
+# End of https://www.gitignore.io/api/django
+docker-compose.yaml
+docker-compose.yml
```
Comment (mjpieters) [inline server/.gitignore]: Neither should this. This. .gitigtore could just be merged with the root .gitignore anyway.

File: server/botany/settings.py [hunk 3]
```diff
@@ -104,7 +104,17 @@
...
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': 'botany',
+        'USER': os.getenv("POSTGRES_USER"),
+        'PASSWORD': os.getenv("POSTGRES_PASSWORD"),
+        'HOST': os.getenv("POSTGRES_URL"),
+        'PORT': '5432',
+    }
+}
```
Comment (mjpieters) [inline server/botany/settings.py]: Should be moved to a different pull req too.

File: server/botany/settings.py [hunk 4]
```diff
@@ -156,13 +166,3 @@
...
-# Raven
-
-RAVEN_PROJECT = os.getenv("RAVEN_PROJECT")
-RAVEN_PUBLIC_KEY = os.getenv("RAVEN_PUBLIC_KEY")
-RAVEN_PRIVATE_KEY = os.getenv("RAVEN_PRIVATE_KEY")
-
-if RAVEN_PROJECT is not None:
-    RAVEN_CONFIG = {
-        "dsn": f"https://{RAVEN_PUBLIC_KEY}:{RAVEN_PRIVATE_KEY}@sentry.io/{RAVEN_PROJECT}"
-    }
```
Comment (mjpieters) [inline server/botany/settings.py]: Ditto

File: server/botany/views.py [hunk 5]
```diff
@@ -96,6 +96,30 @@ def bot_head_to_head(request, bot_id, other_bot_id):
...
+    try:
+        bot_move, state = rv
+        print("Got a state")
+    except TypeError:
+        bot_move, state = rv, None
+        print("Got no state")
+    move_list.append(bot_move)
+    boards = runner.rerun_game(game_mod, move_list)
+    board = boards[-1]
+    return (board,move_list)
```
Comment (mjpieters) [inline server/botany/views.py]: Could you run the `fixup.sh` file at the root to format the code with `black`?

File: server/botany/views.py [hunk 6]
```diff
@@ -129,31 +153,7 @@ def play(request, bot1_id, bot2_id):
...
-
-            rv = fn(**args)
-            try:
-                next_next_move, state = rv
-                print("Got a state")
-            except TypeError:
-                next_next_move, state = rv, None
-                print("Got no state")
-            move_list.append(next_next_move)
+            board, move_list = _make_bot_move(game_mod,bot,board,move_list,other_token,state)
```
Comment (mjpieters) [inline server/botany/views.py]: You completely dropped state handling here. The original code would parse out the state with `json.loads(b64decode(request.POST["state"]))`. This breaks bots that keep a state.

File: server/botany/views.py [hunk 7]
```diff
@@ -97,6 +96,29 @@ def bot_head_to_head(request, bot_id, other_bot_id):
     return render(request, "botany/bot_head_to_head.html", ctx)
 
 
+def _make_bot_move(game_mod, bot_object, board, move_list, token, state):
+    bot_mod = loader.create_module_from_str("bot", bot_object.code)
+    fn = bot_mod.get_next_move
+    param_list = runner.get_param_list(fn)
+    all_args = {"board": board, "move_list": [], "token": token, "state": state}
+    args = {param: value for param, value in all_args.items() if param in param_list}
+    print(args)
```
Comment (inglesp) [inline server/botany/views.py]: Could you remove this `print()`, and the ones immediately below?

**Comment by percevalve** [comment]
Let me re-submit this PR from a clean branch.

**Pull request closed without merging**